components/builder-worker/src/runner/mod.rs

```diff
@@ -703,26 +738,31 @@ impl RunnerMgr {
                 }
             }
 
-            let res = rx.try_recv();
+            let res = rx.try_next();
 
             if let Ok(r) = res {
-                let job: Job = r;
+                let job: Job = r.unwrap();
```

Comment:
We could make this `if let Ok(Some(job)) = res {` to avoid the need to unwrap `r`

Reply:
Addressed in a33367f0febdd930dfb5cba3892af203fc5b3a51